compiler/x/amd64/codegen/OMRMemoryReference.cpp

```diff
@@ -400,10 +400,26 @@ OMR::X86::AMD64::MemoryReference::addMetaDataForCodeAddressWithLoad(
          if (sr.getSymbol()->isStatic())
             {
             if (cg->needClassAndMethodPointerRelocations())
-               cg->addExternalRelocation(new (cg->trHeapMemory()) TR::ExternalRelocation(displacementLocation, (uint8_t *)srCopy,
-                                                                                         (uint8_t *)(uintptr_t)containingInstruction->getNode()->getInlinedSiteIndex(),
-                                                                                         TR_ClassAddress, cg),__FILE__, __LINE__,
-                                                                                         containingInstruction->getNode());
+               {
+               if (cg->comp()->getOption(TR_UseSymbolValidationManager))
```

Comment:
Wouldn't these relocation changes also be needed for other platforms?

Reply:
Yes they would, I just didn't have the time to develop and test this on all platforms; for now this code only works on x86-64. The reason being that I only added a `symbolFromManager` relo type that is used to patch a pointer; however, this isn't sufficient on a platform like P that patches pointers differently depending on whether the relo record was `TR_ClassAddress` or `TR_ClassPointer`. In the next few months I have to make all of this work on all platforms.